docs/versioned_docs/version-v3.2.0/02-Features/running-integration-tests.md

```diff
@@ -0,0 +1,169 @@
+{
+    // Setting up the Azure Service Bus consumer host...
+    private readonly ServiceBusEventConsumerHost _serviceBusEventConsumerHost;
+
+    [Fact]
+    public async Task Publish_ValidParameters_Succeeds()
+    {
+        // Publishing the event...
+
+        // Receiving an event based on it's ID (uses default exponential back-off with 5 retries).
+        _serviceBusEventConsumerHost.GetReceivedEvent("<your-event-id>");
+
+        // Receiving an event based on it's ID, using custom retry count with exponential back-off.
+        _serviceBusConsumerHost.GetReceivedEvent("<your-event-id>", retryCount: 7);
+
+        // Receiving an event based it's ID, using a time-out.
+        _serviceBusConsumerHost.GetReceivedEvent("<your-event-id>", timeout: TimeSpan.FromSeconds(30));
+
+        // Receiving an event based on if it's an `CloudEvent` event, using a time-out.
+        _serviceBusConsumerHost.GetReceivedEvent((CloudEvent cloudEvent) => cloudEvent.Subject == "Order", timeout: TimeSpan.FromSeconds(30));
+
+        // Receiving an event based on if it's an `EventGridEvent` event, using a time-out.
+        _serviceBusConsumerHost.GetReceivedEvent((EventGridEvent eventGridEvent) => eventGridEvent.Subject == "Order", timeout: TimeSpan.FromSeconds(30));
+
+        // Receiving an event based on it's event payload (can either be an `CloudEvent` or an `EventGridEvent`), using a time-out.
+        _serviceBusConsumerHost.GetReceivedEvent((OrderEventData orderData) => orderData.OrderId == "<your-order-id>", timeout: TimeSpan.FromSeconds(30));
+    }
+}
+```
+
+### Azure infrastructure
+
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Farcus-azure%2Farcus.eventgrid%2Fmaster%2Fdeploy%2Farm%2Ftesting-infrastructure%2Fazuredeploy.json" target="_blank">
+    <img src="https://azuredeploy.net/deploybutton.png"/>
+</a>
+<a href="http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2Farcus-azure%2Farcus.eventgrid%2Fmaster%2Fdeploy%2Farm%2Ftesting-infrastructure%2Fazuredeploy.json" target="_blank">
+    <img src="./../media/logos/armviz.png"/>
+</a>
+
+
+When running integration tests with Azure, Arcus needs to be one of the consumers of your custom Azure Event Grid Topic. By doing this, it will send all received events to an [Azure Service Bus Topics](https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-messaging-overview#topics) on which every test run will create a subscription to poll for new events.
+
+The `ServiceBusEventConsumerHost` will poll for new messages on the subscription that was created and store them internally. Once the host closes, it will shut itself down and delete the subscription on the topic.
+
+In the future, we will provide an ARM template which allows you to easily deploy an Azure Logic App that will receive all events from Event Grid and store them on the Service Bus Topic, but you can also provide your own flow to achieve this.
```

Comment:
I'd prefer a bicep template instead of ARM.   Not that ARM is going to disappear (bicep compiles to ARM), but bicep is much easier and more developer friendly :)

Reply:
We can have both, 😄 I forgot about this. This is maybe something we can come up with for other libraries as well. Some do need some Azure setup (like background jobs, messaging...) and maybe ARM/bicep templates can help here.

Reply:
I'm sure it will.   We can then maybe in some cases even use those templates in our own pipelines for running our tests

Reply:
Added it here: https://github.com/arcus-azure/arcus/issues/215 so we can track this.